presto-docs/src/main/sphinx/admin/dynamic-filtering.rst

```diff
@@ -0,0 +1,219 @@
+for dynamic filtering may incur additional CPU overhead during query execution.
+Therefore, to limit the overheads of collecting dynamic filters
+to the cases where the join operator is likely to be selective,
+Presto defines thresholds on size of dynamic filters collected from build side tasks.
+Collection of dynamic filters for joins with large build sides can be enabled
+using the ``enable-large-dynamic-filters`` configuration property or the
+``enable_large_dynamic_filters`` session property.
+
+When large dynamic filters are enabled, limits on size of dynamic filters can be configured
+for each join distribution type using the configuration properties
+``dynamic-filtering.large-broadcast.max-distinct-values-per-driver``,
+``dynamic-filtering.large-broadcast.max-size-per-driver`` and
+``dynamic-filtering.large-broadcast.range-row-limit-per-driver`` and their equivalent
+for partitioned join distribution type.
+
+Similarly, limits for dynamic filters when ``enable-large-dynamic-filters`` is not enabled
+can be configured using configuration properties like
+``dynamic-filtering.large-partitioned.max-distinct-values-per-driver``,
+``dynamic-filtering.large-partitioned.max-size-per-driver`` and
+``dynamic-filtering.large-partitioned.range-row-limit-per-driver`` and their equivalent
+for broadcast join distribution type.
+
+The properties based on ``max-distinct-values-per-driver`` and ``max-size-per-driver``
+define thresholds for the size up to which dynamic filters are collected in a
+distinct values data structure. When the build side exceeds these thresholds,
+Presto switches to collecting min and max values per column to reduce overheads.
+This min-max filter has much lower granularity than the distinct values filter.
+However, it may still be beneficial in filtering some data from the probe side,
+especially when a range of values is selected from the build side of join.
+The limits for min-max filters collection are defined by the properties
+based on ``range-row-limit-per-driver``.
+
+Data layout considerations
+--------------------------
+Dynamic filtering is most effective when a connector stores data in a layout that can
+be leveraged for efficient filtering at runtime.
+For Hive connector, dynamic filter predicates are pushed into the ORC and Parquet readers
+on the worker nodes to perform stripe or row-group pruning and save on disk I/O.
+Sorting files by the columns used in join criteria significantly improves the
+effectiveness of stripe or row-group pruning. This is because sorting results in similar
+data getting stored within the same stripe or row-group which greatly improves the
+selectivity of the min/max indexes maintained at stripe or row-group level.
+
+Partitioned Hive tables stored in any data format benefit from the
+dynamic partition pruning optimization for broadcast as well as partitioned joins.
+
+Limitations
+-----------
+* Dynamic filtering is currently implemented only for Hive and Memory connectors
```

Comment:
I think we should remove that and add it to the connectors instead ..

Reply:
I've added links to the connector docs. I think it's good to have this point noted in one place here.